tests/robot-tests/tests/libs/utilities.py

```diff
@@ -163,31 +163,40 @@ def parent_does_not_contain_matching_elements() -> bool:
         raise_assertion_error(err)
 
 
-def get_child_element(parent_locator: object, child_locator: str):
-    try:
-        children = get_child_elements(parent_locator, child_locator)
-
-        if len(children) == 0:
-            raise_assertion_error(
-                f"Found no elements matching child locator {child_locator} under parent "
-                f"locator {parent_locator} in utilities.py#get_child_element()"
-            )
-
-        if len(children) > 1:
-            logger.warn(
-                f"Found {len(children)} child elements matching child locator {child_locator} "
-                f"under parent locator {parent_locator} in utilities.py#get_child_element() - "
-                f"was expecting only one. Consider making the parent selector more specific. "
-                f"Returning the first element found."
-            )
+def get_child_element(parent_locator: object, child_locator: str, retries: int = 5, delay: float = 1.0):
+    def log_and_raise(message: str):
+        logger.warning(message)
+        raise AssertionError(message)
 
-        return children[0]
-    except Exception as err:
-        logger.warn(
-            f"Error whilst executing utilities.py get_child_element() with parent {parent_locator} and child "
-            f"locator {child_locator} - {err}"
-        )
-        raise_assertion_error(err)
+    for attempt in range(retries):
+        try:
+            children = get_child_elements(parent_locator, child_locator)
+            if not children:
+                if attempt < retries - 1:
+                    logger.info(f"Retrying... ({attempt + 1}/{retries})")
+                    time.sleep(delay)
+                    continue
+                else:
+                    log_and_raise(
+                        f"No elements matching child locator '{child_locator}' under parent locator '{parent_locator}' after {retries} retries"
+                    )
+            if len(children) > 1:
+                logger.warning(
+                    f"Multiple ({len(children)}) elements found for child locator '{child_locator}' under parent locator '{parent_locator}'. "
+                    f"Returning the first element. Consider refining the parent selector."
+                )
+            return children[0]
+
+        except Exception as err:
+            if attempt < retries - 1:
+                logger.info(f"Retrying due to error... ({attempt + 1}/{retries})")
+                time.sleep(delay)
+                continue
+            else:
+                logger.warning(
+                    f"Error in get_child_element() with parent '{parent_locator}' and child locator '{child_locator}': {err}"
+                )
+                raise AssertionError(err)
```

Comment:
Could use `log_and_raise` here as well.

Reply:
I've pulled out `log_and_raise` by making use of existing method  ` raise_assertion_error`